pilot/pkg/features/pilot.go

```diff
@@ -671,6 +671,11 @@ var (
 		"Metric scope rotation interval, set to 0 to disable the metric scope rotation").Get()
 	MetricGracefulDeletionInterval = env.Register("METRIC_GRACEFUL_DELETION_INTERVAL", 5*time.Minute,
 		"Metric expiry graceful deletion interval. No-op if METRIC_ROTATION_INTERVAL is disabled.").Get()
+
+	SelfSignedAlgorithm = env.Register("CITADEL_SELF_SIGNED_CA_CERT_ALGORITHM", "RSA",
+		"The algorithm type of the self-signed CA cert created by istiod. Only RSA or ECC is supported. RSA is the default value").Get()
+	EccSigAlgEnv = env.Register("ECC_SIGNATURE_ALGORITHM", "", "The type of ECC signature algorithm to use when generating private keys").Get()
```

Comment:
Would be good to have a default choice of ```EccSigAlgEnv``` if ```SelfSignedAlgorithm``` is set to ECC. Can't this be set to ECDSA?

Reply:
I made a new environment variable only for citadel/pilot to separate it from pilot-agent.